Sources/NIOHTTP1/NIOTypedHTTPServerUpgradeHandler.swift

```diff
@@ -385,17 +391,18 @@ public final class NIOTypedHTTPServerUpgradeHandler<UpgradeResult: Sendable>: Ch
         // then remove ourselves from the pipeline.
         let channel = context.channel
         let pipeline = context.pipeline
-        let loopBoundContext = context.loopBound
-        self.removeExtraHandlers(pipeline: pipeline).flatMap {
-            let context = loopBoundContext.value
-            return self.sendUpgradeResponse(context: context, responseHeaders: responseHeaders)
-        }.flatMap {
-            pipeline.syncOperations.removeHandler(self.httpEncoder)
-        }.flatMap { () -> EventLoopFuture<UpgradeResult> in
-            upgrader.upgrade(channel: channel, upgradeRequest: requestHead)
-        }.hop(to: context.eventLoop)
+
+        self.removeExtraHandlers(pipeline: pipeline)
+            .assumeIsolated()
+            .flatMap {
+                self.sendUpgradeResponse(context: context, responseHeaders: responseHeaders)
+            }.flatMap {
+                pipeline.syncOperations.removeHandler(self.httpEncoder)
+            }.flatMap { () -> EventLoopFuture<UpgradeResult> in
+                upgrader.upgrade(channel: channel, upgradeRequest: requestHead)
+            }.nonisolated().hop(to: context.eventLoop)
+            .assumeIsolated()
```

Comment:
`nonisolated().hop(to:).assumeIsolated()` shouldn't do anything meaningful here. We were already isolated.